src/navigation/mobileStepper/MobileStepper.jsx

```diff
@@ -0,0 +1,184 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { isEmpty } from 'lodash';
+
+const propTypes = {
+    activeStep: PropTypes.number.isRequired,
+    backStepAction: PropTypes.func.isRequired,
+    backStepLabel: PropTypes.string.isRequired,
+    id: PropTypes.string,
+    lastStepAction: PropTypes.func.isRequired,
+    lastStepLabel: PropTypes.string.isRequired,
+    callToAction: PropTypes.shape({
+        text: PropTypes.string,
+        url: PropTypes.string,
+    }),
+    nextStepAction: PropTypes.func.isRequired,
+    nextStepLabel: PropTypes.string.isRequired,
+    steps: PropTypes.number.isRequired,
+};
+
+const defaultProps = {
+    id: 'navigation_mobile_stepper',
+    callToAction: null,
+};
+
+const MobileStepper = ({
+    activeStep,
+    backStepAction,
+    backStepLabel,
+    id,
+    lastStepAction,
+    lastStepLabel,
+    callToAction,
+    nextStepAction,
+    nextStepLabel,
+    steps,
+}) => {
+    const useStyles = makeStyles(({
+        palette,
+    }) => ({
+        mobileStepper: {
+            display: 'grid',
+            gap: 15,
+            gridTemplateColumns: activeStep === 0 ? 'auto' : 'repeat(3, auto)',
+            gridTemplateRows: 'repeat(2, 1fr)',
+            alignItems: 'center',
+            justifyContent: activeStep === 0 ? 'center' : 'space-between',
+            gridTemplateAreas:
+                activeStep === 0 ? `
+                    "nextButton"
+                    "stepperIndicator"
+            ` :
+                    `
+                "backButton callToAction nextButton"
+                "stepperIndicator stepperIndicator stepperIndicator"
+            `,
+        },
+        stepperIndicator: {
+            alignItems: 'center',
+            display: 'flex',
+            gridArea: 'stepperIndicator',
+            justifyContent: 'center',
+        },
+        step: {
+            background: palette.grey[300],
+            borderRadius: 99,
+            display: 'inline-block',
+            height: 8,
+            marginRight: 5,
+            width: 8,
+        },
+        stepActive: {
+            animation: '$fadeIn 250ms ease-in-out',
+            background: palette.active.primary,
+        },
+        backButton: {
+            background: palette.grey[400],
+        },
+        nextButton: {
+            gridArea: 'nextButton',
+        },
+        '@keyframes fadeIn': {
+            '0%': {
+                opacity: 0,
+                transform: 'scale(0)',
+            },
+            '50%': {
+                opacity: 1,
+                transform: 'scale(1.4)',
+            },
+            '100%': {
+                opacity: 1,
+                transform: 'scale(1)',
+            },
+        },
+    }));
+
+    const classes = useStyles();
+
+    const backButton =
+        activeStep > 0 && steps - 1 ? (
+            <Button
+                className={classes.backButton}
+                id={`${id}--button_back`}
```

Comment:
The `id` should be used in the modifier pattern. I think they should be `${BEM_MOBILE_STEPPER}--button_back-${id}`